src/ashley/machina_extensions/forum/views.py

```diff
@@ -5,34 +5,52 @@
     This module overrides views provided by the ``forum`` application.
 
 """
+from django.db.models import F
 from django.http import Http404
 from django.urls import reverse
 from django.utils.http import urlencode
 from django.views.generic import UpdateView
 from machina.apps.forum.views import ForumView as BaseForumView
+from machina.apps.forum.views import IndexView as BaseIndexView
 from machina.apps.forum_permission.viewmixins import (
     PermissionRequiredMixin as BasePermissionRequiredMixin,
 )
 from machina.core.db.models import get_model
 from machina.core.loading import get_class
 
 Forum = get_model("forum", "Forum")
-
+ForumVisibilityContentTree = get_class("forum.visibility", "ForumVisibilityContentTree")
 
 ORDER_VAR = "o"
 PermissionRequiredMixin: BasePermissionRequiredMixin = get_class(
     "forum_permission.viewmixins", "PermissionRequiredMixin"
 )
 
 
-class ForumView(BaseForumView):  # pylint: disable=too-many-ancestors
-    """Displays a forum and its topics."""
+class OrderByColumnMixin:
+    """Helper class to order by columns"""
 
-    # header columns
-    list_display = ["subject", "posts_count", "views_count", "last_post_on"]
     # default ordering
     default_ordering = "-last_post_on"
 
+    def __init__(self):
+        """Init called with default_ordering and list_display"""
+        super().__init__()
+        self.params = {}
+
+    def get(self, request, *args, **kwargs):
+        """Handles GET requests."""
+        # collect the current params for the order by in request
+        self.params = dict(request.GET.items())
+        try:
+            super_get = getattr(super(), "get")
+            return super_get(request, *args, **kwargs)
+        except Exception as exception:
```

Comment:
I think you should catch only the precise exception you're expecting

Reply:
j'ai mis except AttributeError as exception: